libs/libvtrutil/src/vtr_thread_pool.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+/** 
+ * @file RouterThreadPool.h
+ * @brief A generic thread pool for parallel task execution
+ */
+
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <condition_variable>
+#include <memory>
+#include <atomic>
+#include <functional>
+#include <cstddef>
+#include <vector>
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+namespace vtr {
+
+class thread_pool {
+private:
+    struct ThreadData {
+        std::thread thread;
+        std::queue<std::function<void()>> task_queue;
+        std::mutex queue_mutex;
+        std::condition_variable cv;
+        bool stop = false;
+        size_t thread_id;  // For debugging
+        size_t tasks_completed{0};  // Track number of completed tasks
+
+        ThreadData(size_t id) : thread_id(id) {}
+    };
+
+    std::vector<std::unique_ptr<ThreadData>> threads;
+    std::atomic<size_t> next_thread{0};  // For round-robin assignment
+    std::atomic<size_t> total_tasks_queued{0};
+    vtr::Timer pool_timer;  // Track pool lifetime
+    std::atomic<size_t> active_tasks{0};
+    std::mutex completion_mutex;
```

Comment:
Comment these.  (all the variables).  
From our discussion: completion_mutex is to wait for all the tasks in the queue to finish.